src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerTerminationValidator.java

```diff
@@ -54,16 +55,32 @@ public ApiErrors validate(HttpServletRequest request, OfficerFilingDto dto, Stri
 
         // Retrieve data objects required for the validation process
         final Transaction transaction = transactionService.getTransaction(transId, passthroughHeader);
-        final AppointmentFullRecordAPI companyAppointment = companyAppointmentService.getCompanyAppointment(transaction.getCompanyNumber(), dto.getReferenceAppointmentId(), passthroughHeader);
+        final AppointmentFullRecordAPI companyAppointment = validateOfficerIsIdentified(request, errorList, dto, transaction, passthroughHeader);
         final CompanyProfileApi companyProfile = companyProfileService.getCompanyProfile(transId, transaction.getCompanyNumber(), passthroughHeader);
 
-        // Perform validation
-        validateMinResignationDate(request, errorList, dto);
-        validateTerminationDateAfterIncorporationDate(request, errorList, dto, companyProfile, companyAppointment);
+        if (companyAppointment != null) {
+            // Perform validation
+            validateMinResignationDate(request, errorList, dto);
+            validateTerminationDateAfterIncorporationDate(request, errorList, dto, companyProfile, companyAppointment);
+        }
 
         return new ApiErrors(errorList);
     }
 
+    public AppointmentFullRecordAPI validateOfficerIsIdentified(HttpServletRequest request, List<ApiError> errorList, OfficerFilingDto dto, Transaction transaction, String passthroughHeader) {
```

Comment:
I think the name of this method could be a bit clearer in showing that it is fetching the company appointment from a service.